src/cursor/cur_std.c

```diff
@@ -1176,42 +1176,53 @@ __wt_cursor_largest_key(WT_CURSOR *cursor)
 int
 __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
 {
-    WT_CONFIG_ITEM cval;
     WT_CURSOR_BTREE *cbt;
     WT_DECL_RET;
     WT_ITEM key;
     WT_SESSION_IMPL *session;
     int exact;
+    char *cfg1, *cfg2;
     bool inclusive;
 
+    cfg1 = cfg2 = NULL;
     cbt = (WT_CURSOR_BTREE *)cursor;
     exact = 0;
-    inclusive = false;
+    inclusive = true;
 
-    CURSOR_API_CALL_CONF(cursor, session, bound, config, cfg, NULL);
+    CURSOR_API_CALL(cursor, session, bound, NULL);
 
     if (CUR2BT(cursor)->type == BTREE_COL_FIX)
         WT_ERR_MSG(session, EINVAL, "setting bounds is not compatible with fixed column store.");
 
     if (F_ISSET(cursor, WT_CURSTD_PREFIX_SEARCH))
         WT_ERR_MSG(session, EINVAL, "setting bounds is not compatible with prefix search.");
 
-    WT_ERR(__wt_config_gets(session, cfg, "action", &cval));
-    if (WT_STRING_MATCH("set", cval.str, cval.len)) {
-        WT_ERR(__wt_config_gets(session, cfg, "inclusive", &cval));
-        inclusive = cval.val != 0;
+    if (config == NULL || config[0] == '\0')
+        WT_ERR_MSG(session, EINVAL, "an empty config is not valid when setting bounds");
 
-        /* Check that bound is set with action set configuration. */
-        WT_ERR(__wt_config_gets(session, cfg, "bound", &cval));
-        if (cval.len == 0)
-            WT_ERR_MSG(session, EINVAL, "setting bounds must require the bound configuration set");
+    cfg1 = strstr(config, "action");
+    if (cfg1 == NULL)
+        WT_ERR_MSG(session, EINVAL,
+          "an action must be specified when setting bounds, either \"clear\" or \"set\"");
```

Comment:
Thanks for the suggestion Don! I agree it would be better if we codify it a little. I'll keep it in `cur_std` as ideally we don't want this being used in a lot of places. Perhaps in the future the config parsing project will resolve the need for this change.

I added your suggestion in: [bf822bd](https://github.com/wiredtiger/wiredtiger/pull/8342/commits/bf822bdefd80d446284c0cf3bda18dd5c7bb4010)